src/sentry/static/sentry/app/components/modals/addDashboardWidgetModal.tsx

```diff
@@ -117,6 +121,11 @@ function normalizeQueries(
       fields = fields.slice(0, 3);
     }
 
+    if (isTimeseriesChart || displayType === 'world_map') {
+      // Filter out fields that will not generate numeric output types
+      fields = fields.filter(field => isLegalYAxisType(aggregateOutputType(field)));
+    }
```

Comment:
Should this `filter` be applied prior to the `slice` above so we don't go below 3 fields where possible?

Reply:
I'll move it to be earlier.